### PR TITLE
8323515: Create test alias "all" for all test roots

### PR DESCRIPTION
**Pull request by shipilev**
Since recent work to improve `tier4` performance, we actually test `tier{1,2,3,4}` often, which includes all the tests in current tree. It would be more convenient to just have the `all` test group/alias, so that we can do `make test TEST=all`. This also gives a parallelism / run time benefit, as we do not wait for tests in each tier to complete before moving to next tier. 

Sample run on out-of-the-box Linux x86_64 fastdebug is below. For some environments one also needs to supply a few keywords like `!printer` to skip tests that cannot complete without failure due to misconfiguration. I left the keywords as is to show how would a failing run look. There is also an existing shortcut in build system that allows to run this with `make test-all`.

```
% make test TEST=all

Test selection 'all', will run:
* jtreg:test/hotspot/jtreg:all
* jtreg:test/jdk:all
* jtreg:test/langtools:all
* jtreg:test/jaxp:all
* jtreg:test/lib-test:all

(...about 6 hours later...)

==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
>> jtreg:test/hotspot/jtreg:all                       6731  6702    29     0 <<
>> jtreg:test/jdk:all                                 9962  9951    11     0 <<
   jtreg:test/langtools:all                           4469  4469     0     0   
   jtreg:test/jaxp:all                                 513   513     0     0   
   jtreg:test/lib-test:all                              32    32     0     0   
==============================
TEST FAILURE
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8323515](https://bugs.openjdk.org/browse/JDK-8323515): Create test alias "all" for all test roots (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [7f6797b6](https://git.openjdk.org/jdk/pull/17422/files/7f6797b6330d6dc38395854317c4cb2cad050f1f)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17422/head:pull/17422` \
`$ git checkout pull/17422`

Update a local copy of the PR: \
`$ git checkout pull/17422` \
`$ git pull https://git.openjdk.org/jdk.git pull/17422/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17422`

View PR using the GUI difftool: \
`$ git pr show -t 17422`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17422.diff">https://git.openjdk.org/jdk/pull/17422.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17422#issuecomment-1892453279)

### CODE REVIEW

File: test/jdk/TEST.groups [hunk 1]
```diff
@@ -20,6 +20,14 @@
 #  questions.
 #
 
+###############################################################################
+#
+# Catch-all definitions
+#
+
+all = \
```
Comment (dholmes-ora) [inline test/jdk/TEST.groups:28]: Why no `jdk_all` definition in this case?

**Review by dholmes-ora — APPROVED** [review]
Okay - change is harmless with no ongoing maintenance cost.

File: test/jdk/TEST.groups (hunk 1 above)
Comment (shipilev) [reply test/jdk/TEST.groups:28]: Tried not to introduce new `*_all` groups here. `jdk_all` would be the same as `jdk:all`, TBH. But we still can do it for symmetry reasons, see new commit.

Comment (AlanBateman) [reply test/jdk/TEST.groups:28]: "all" looks okay but the comment "Catch-all" suggests something else, shouldn't be "All tests"?

Comment (shipilev) [reply test/jdk/TEST.groups:28]: Yeah, we can do "All tests" instead. See new commit.

**Review by AlanBateman — APPROVED** [review]

**Comment by shipilev** [comment]
Any other reviews needed for this? Nominally, this changes the test groups in langtools, so maybe @lahodaj or @biboudis want to take a look. For jaxp, @JoeWang-Java, maybe?

**Review by JoeWang-Java — APPROVED** [review]
Thanks for the reminder. The new alias is nice, easier to run all tier tests. I often run xml-only as well that includes jaxp_all plus a small set of jaxp tests in jdk_all (test/jdk/javax/xml/jaxp). But that's just me, jdk_all already covers those tests.

**Comment by lahodaj** [comment]
I have nothing against it, but I don't think I know enough about details of jtreg configuration to provide approval. OTOH, I personally don't see a strong need for it.

**Comment by shipilev** [comment]
All right, thanks! @lmesnik, I realized I forgot to ask if you had objections to this.

**Review by lmesnik — APPROVED** [review]

**Comment by shipilev** [comment]
Thank you all!

/integrate

**Comment by DamonFool** [comment]
Hi @shipilev , plese see https://github.com/openjdk/jdk/pull/17558 .
Thanks.

**Pull request closed without merging**